### ISSUE TITLE
[Bug]: Network setting do not apply on apps created from templates

### ISSUE THREAD

**Issue by mr47** [coollabsio/coolify#4276]
### Error Message and Logs

Creating grafana for example (no modification was done to template) exept
![Image](https://github.com/user-attachments/assets/ba22b2ea-e5dd-4bed-9a2c-8901c0ceac0a)
Docker-compose (deployable)
![Image](https://github.com/user-attachments/assets/2026ca4d-7758-41d0-8de9-3a68700b58fd)

Setting option connect to predefined networks have no effect. (its already set)

### Steps to Reproduce

1. Create a grafana app
2. Add coolify network by hand using `edit` / setting `connect to predefined network`
3. Check deployable `docker-compose.yaml`
4. Observe absence of networks

PS: Maybe its merging object problem ?

### Coolify Version

v4.0.0-beta.367

### Are you using Coolify Cloud?

No (self-hosted)

### Operating System and Version (self-hosted)

TrueNas ElectricEel-24.10.0.2

**Comment by djsisson** [comment]
@mr47 after ticking connect to predefined network and redeploying, are you saying if you run docker network inspect coolify you dont see the container?

**Comment by mr47** [comment]
@djsisson yes, `docker-compose.yaml` deployable does not contain coolify as network

**Comment by djsisson** [comment]
@mr47 it doesnt add it to compose, it manaully adds it to the network

**Comment by djsisson** [comment]
@mr47  you need to add a top level network section if thats what you want, it does work when adding your own networks, see appwrite template for an example

**Comment by mr47** [comment]
@djsisson that's the issue coolify not sync docker compose with ui, so this behavior is magical.

Even if I add
```yaml
networks:
   mynet: 
     name: myned
     external: true
services: 
    myService:
       network: 
          mynet: null
```

That will result in deployable docker-compose without network `mynet`

**Comment by djsisson** [comment]
@mr47  can you paste link to your compose please

**Comment by ShadowArcanist** [comment]
Closing this due to inactivity from the issue author for over a year. Please comment here if this is still an issue.
